src/types.ts

```diff
@@ -12,6 +12,10 @@ export interface PromiseResultShape<T, E> {
     mapErr: <U>(fn: (err: E) => U) => PromiseResultShape<T, U>;
     get: () => T;
     getOr: (orValue: T) => T;
+    onResolved: <U>(fn: (value: T) => U) => PromiseResultShape<T, E>;
```

Comment:
Actually the functions don't have to be generic, because the `U` type won't be used anywhere, we cannot get the return value of `fn` in any way. I'd just use `unknown` or `any` here.

Reply:
or `any`? wat? :D